devfile/devfile-web
# Run linter checks when ready to push (not when committing)

## What does this PR do / why we need it
While trying to address https://github.com/devfile/devfile-web/pull/97#pullrequestreview-1640165353, I noticed that the current pre-commit hook might take too many seconds to run the linter checks.
And running those checks every time someone wants to commit actually adds friction to the developer experience.
IMO, developers should feel free to commit frequently while iterating on their local branches.
I generally tend to commit a lot when working on a branch. Then when my branch is ready, I can rewrite the history before pushing.
To this end, it makes sense to me to defer running such checks until the developer is ready to push.

## Which issue(s) does this PR fix
&mdash;

## PR acceptance criteria

_Update the [sidebar](https://github.com/devfile/devfile-web/tree/main/apps/landing-page#configuring-navigation) if there is a new file added or an existing filename is changed_

## How to test changes / Special notes to the reviewer
No linter checks after each local commit. The pre-push hook should be executed only when running `git push`
```
$ git commit -am "wip"                                                                                                                                                              
[test 8a8ed69] wip                                                                         
 1 file changed, 0 insertions(+), 0 deletions(-)
[...]

$ git push -u origin wip                                                                                                                                                                              
                                                                                                                                                                                               
 >  NX   workspace.json is ignored                                                                                                                                                             
                                                                                                                                                                                               
   Nx no longer reads configuration from workspace.json.                                                                                                                                       
   Run "nx g @nrwl/workspace:fix-configuration" to split workspace.json into individual project.json files.                                                                                    
                                                                                                                                                                                               
                                                                                                                                                                                               
 >  NX   Affected criteria defaulted to --base=main --head=HEAD                                                                                                                                
                                                                                                                                                                                               
                                                                                                                                                                                               
    ✔  nx run landing-page:lint (15s)                                                                                                                                                          
    ✔  nx run registry-viewer:lint (15s)                                                                                                                                                       
    ✔  nx run core:lint (26s)                                                                                                                                                                  
    ✔  nx run registry-viewer-e2e:lint (15s)                                                                                                                                                   
    ✔  nx run landing-page-e2e:lint (15s)                                                                                                                                                      
                                                                                                                                                                                               
 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————— 
                                                                                                                                                                                               
 >  NX   Successfully ran target lint for 5 projects (30s)                                                                                                                                     
                                                                                                                                                                                               
   View logs and investigate cache misses at https://nx.app/runs/dtSGEQIM4u                                                                                                                    
                                                                                                                                                                                               
libs/core/src/components/hero/hero.tsx                                                                                                                                                         
husky - pre-push hook exited with code 1 (error)                                                                                                                                               
error: failed to push some refs to 'github.com:rm3l/devfile-web.git' 
```
